dex/src/matching.rs

```diff
@@ -466,29 +478,31 @@ impl<'ob> OrderBookState<'ob> {
                 owner_slot: best_bid_ref.owner_slot(),
                 fee_tier: maker_fee_tier,
                 client_order_id: NonZeroU64::new(best_bid_ref.client_order_id()),
-            });
+            };
             event_q
-                .push_back(maker_fill)
+                .push_back(Event::new(maker_fill))
                 .map_err(|_| DexErrorCode::EventQueueFull)?;
-
+            emit!(maker_fill.to_fill_event()?);
```

Comment:
Would it be acceptable to move all event emissions to consume_events? It's possible this breaks existing composers that are already running tight on compute units.